### PR TITLE
SA-2209 - (UI) Add missing unit tests for global Analytics Report components

### PR DESCRIPTION
**Pull request by nicklemmon**
[SA-2209](https://sparkpost.atlassian.net/browse/SA-2209)

### What Changed

- Adds unit tests for global Analytics Report components
- [Renames a click handler prop](https://github.com/SparkPost/2web2ui/pull/2091/commits/5548fc756fea22c5bc880bcc14eeb353fa541f76) for the sake of clarity
- Upgrades React Testing Library dependencies (thought it was worth doing):
    - `@testing-library/react`
    - `@testing-library/react-hooks`
    - `@testing-library/user-event`
    - `@testing-library/jest-dom`
- Updates an existing test that began to fail after updating `@testing-library/user-event`. Looks like some of the element attributes are set incorrectly - I filed [a follow-up ticket to address](https://sparkpost.atlassian.net/browse/FE-1386) this and other similar issues in the app related to the use of Downshift 

### How To Test

- Verify tests accurately test code paths within each component
- Verify tests pass in CI

### To Do

- [ ] Incorporate feedback

### CODE REVIEW

File: src/components/reportBuilder/Charts.js [hunk 1]
```diff
@@ -79,7 +82,7 @@ export function ChartGroups(props) {
             </Panel.Action>
           </Panel.Header>
 
-          <IndustryModal
+          <IndustryBenchmarkModal
```
Comment (nicklemmon) [inline src/components/reportBuilder/Charts.js:85]: I renamed this to match the name of the file - thought it made things slightly easier to understand.

File: src/components/reportBuilder/AggregatedMetrics.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
 import React, { useMemo } from 'react';
-import { Button, Box, Grid, Text, Inline, LabelValue } from 'src/components/matchbox';
+import styled from 'styled-components';
 import { FilterAlt } from '@sparkpost/matchbox-icons';
+import { Button, Box, Grid, Text, Inline, LabelValue } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/components/reportBuilder/AggregatedMetrics.js:4]: Just moving vendor imports to the top

**Comment by nicklemmon** [comment]
@PatrickSison I ready for another pass - I added a few more test cases to `Charts.test.js` and also renamed the `IndustryModal` to `IndustryBenchmarkModal` to match the filename. I think that was a mistake maybe? Not sure.

Let me know what questions you have.

File: src/components/reportBuilder/Charts.js [hunk 3]
```diff
@@ -16,16 +18,16 @@ import {
 import { ApiErrorBanner } from 'src/components';
 import Loading from 'src/components/loading/PanelLoading';
 import { Heading } from 'src/components/text';
+import { useIndustryBenchmark } from 'src/hooks/reportBuilder';
 import CustomTooltip from './Tooltip';
-import useIndustryBenchmark from 'src/hooks/reportBuilder/useIndustryBenchmark';
-import { INDUSTRY_BENCHMARK_INDUSTRIES } from 'src/constants';
-import { INDUSTRY_BENCHMARK_METRICS_MAP } from 'src/config/metrics';
-import { IndustryModal } from 'src/pages/reportBuilder/components/IndustryBenchmarkModal';
+import { IndustryBenchmarkModal } from 'src/pages/reportBuilder/components';
```
Comment (PatrickSison) [inline src/components/reportBuilder/Charts.js:23]: Good rename

File: src/components/reportBuilder/tests/Charts.test.js [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+        reportOptions: {
+          metrics: DEFAULT_METRICS,
+          comparisons: [],
+        },
+      });
+
+      expect(document.querySelector(CHART_SELECTOR)).toBeInTheDocument();
+    });
+
+    it('renders multiple charts when a rate metric and count metric are active simultaneously', () => {
```
Comment (PatrickSison) [inline src/components/reportBuilder/tests/Charts.test.js:176]: Nice. Thanks for adding this in!

**Review by PatrickSison — APPROVED** [review]
Nice. Rename makes sense too. Still feels a bit weird with how the modal is injected into the charts, but not sure how we can better organize it. Also unrelated to this PR

**Pull request merged**